internal/gatewayapi/runner/runner.go

```diff
@@ -182,6 +203,153 @@ func (r *Runner) deleteAllIRKeys() {
 	}
 }
 
+type DeletableStatus struct {
+	GatewayStatusKeys   map[types.NamespacedName]bool
+	HTTPRouteStatusKeys map[types.NamespacedName]bool
+	GRPCRouteStatusKeys map[types.NamespacedName]bool
+	TLSRouteStatusKeys  map[types.NamespacedName]bool
+	TCPRouteStatusKeys  map[types.NamespacedName]bool
+	UDPRouteStatusKeys  map[types.NamespacedName]bool
+
+	ClientTrafficPolicyStatusKeys  map[types.NamespacedName]bool
+	BackendTrafficPolicyStatusKeys map[types.NamespacedName]bool
+	SecurityPolicyStatusKeys       map[types.NamespacedName]bool
+	BackendTLSPolicyStatusKeys     map[types.NamespacedName]bool
```

Comment:
nit: can we move `BackendTLSPolicyStatusKeys` near L212, since its an upstream resource like `UDPRouteStatus`